localstack-core/localstack/utils/java.py

```diff
@@ -0,0 +1,120 @@
+"""
+Utilities related to Java runtime.
+"""
+
+import logging
+from os import environ
+from urllib.parse import urlparse
+
+from localstack import config
+from localstack.utils.files import new_tmp_file, rm_rf
+from localstack.utils.run import run
+
+LOG = logging.getLogger(__name__)
+
+
+#
+# Network
+#
+
+
+def java_system_properties_proxy() -> dict[str, str]:
+    """
+    Returns Java system properties for network proxy settings as per LocalStack configuration.
+
+    See: https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html
+    """
+    props = {}
+
+    for scheme, default_port, var in [
+        ("http", "80", config.OUTBOUND_HTTP_PROXY),
+        ("https", "443", config.OUTBOUND_HTTPS_PROXY),
+    ]:
+        if var:
+            netloc = urlparse(var).netloc
+            url = netloc.split(":")
+            if len(url) == 2:
+                hostname, port = url
+            else:
+                hostname, port = url[0], default_port
+
+            props[f"{scheme}.proxyHost"] = hostname
+            props[f"{scheme}.proxyPort"] = port
+
+    return props
+
+
+#
+# SSL
+#
+
+
+def build_trust_store(
+    keytool_path: str, pem_bundle_path: str, env_vars: dict[str, str], store_passwd: str
+) -> str:
+    """
+    Build a TrustStore in JKS format from a PEM certificate bundle.
+
+    :param keytool_path: path to the `keytool` binary.
+    :param pem_bundle_path: path to the PEM bundle.
+    :param env_vars: environment variables passed during `keytool` execution. This should contain JAVA_HOME and other relevant variables.
+    :param store_passwd: store password to use.
+    :return: path to the truststore file.
+    """
+    store_path = new_tmp_file(suffix=".jks")
```

Comment:
Yes, currently it's one trust store used for a given version of Opensearch/Elasticsearch installation, reused for all its plugins.

The JRE and thus `keytool` executable changes based on the Opensearch/Elasticsearch version (we use the bundled JRE), it may make sense not to re-use the trust store across versions.